File: src/main/java/Ex11_List.java
import java.util.Arrays;
import java.util.List;

import static java.util.Collections.emptyList;
import static java.util.Collections.singletonList;

public class Ex11_List {
    public static void main(String[] args) {
        List<String> oneFruit = singletonList("Kiwi"); // Declare singleton list of fruit => Immutable

        List<String> fruits = Arrays.asList("Apple", "Banana", "Orange"); // Declare mutable list of fruits

        List<String> otherFruits = List.of("Avocado", "Ananas", "Cranberry"); // Immutable list of fruits

        List<Object> noFruits = emptyList(); // Declare empty list
    }
}

File: src/main/kotlin/Ex11_List.kt
fun main() {
  /* List are Immutable on Kotlin */
  /* MutableList are mutable on Kotlin */

  val oneFruitMutable: MutableList<String> = mutableListOf("Kiwi") // Mutable singleton list of fruit
  val oneFruitImmutable: List<String> = listOf("Kiwi") // Immutable singleton list of fruit

  val fruits: MutableList<String>  = mutableListOf("Apple", "Banana", "Orange") // Mutable list of fruits

  val otherFruits: List<String> = listOf("Avocado", "Ananas", "Cranberry") // Immutable list of fruits

  val noFruitMutable: MutableList<String> = mutableListOf() // Mutable empty list
  val noFruitImmutable: List<String> = listOf() // Immutable empty list
}